src/gdb_packet.c

```diff
@@ -31,119 +31,220 @@
 
 #include <stdarg.h>
 
+typedef enum packet_state {
+	PACKET_IDLE,
+	PACKET_GDB_CAPTURE,
+	PACKET_GDB_ESCAPE,
+	PACKET_GDB_CHECKSUM_UPPER,
+	PACKET_GDB_CHECKSUM_LOWER,
+	PACKET_GDB_COMPLETE,
+} packet_state_e;
+
+static bool noackmode = false;
+
+void gdb_set_noackmode(bool enable)
+{
+	/* 
+	 * If we were asked to disable no ack mode, and it was previously enabled,
+	 * it might mean we got a paket we determined to be the first of a new
+	 * GDB session, and as such it was not acknoledged (before GDB enabled no ack mode),
+	 * better late than never.
+	 * 
+	 * If we were asked after the connection was terminated, sending the ack have no effect.
+	 */
+	if (enable == false && noackmode == true) {
```

Comment:
```suggestion
	if (!enable && noackmode)
```
and drop the extra braces